sched/group/group_addrenv.c

```diff
@@ -119,15 +146,15 @@ int group_addrenv(FAR struct tcb_s *tcb)
   /* Are we going to change address environments? */
 
   flags = enter_critical_section();
-  if (pid != g_pid_current)
+  if (pid != g_pid_current[cpu])
```

Comment:
Suggestion:
Implement a getter for g_pid_current[this_cpu] and store its value upon entry.

Something like this:

int get_pid_current(void)
{
  int pid = g_pid_current[cpu];
  return pid == IDLE_PROCESS_ID ? INVALID_PROCESS_ID : pid;
}

This way you don't need to initialize the struct